README.md

```diff
@@ -1,126 +1,248 @@
+You will need to know your version of Ubuntu to download the right binary. You can find this by going to Start/Windows/Super key > Settings > About > OS Name, or by running the following in your terminal.
+```bash
+lsb_release -a
+```
 
-### Building From Source
+<details> <summary>Ubuntu derivatives</summary>
 
-Recent Ubuntu LTS releases are the only Linux distributions that we fully support. Other Linux distros and other POSIX operating systems (such as macOS) are tended to on a best-effort basis and may not be full featured. Notable requirements to build are:
-* C++17 compiler and standard library
-* boost 1.67+
-* CMake 3.8+
-* (for Linux only) LLVM 7 - 11 (newer versions do not work)
+We currently only support Ubuntu but, if you want to try running our software on an Ubuntu derivative such as Linux Mint, you can determine the version of Ubuntu it was built from using the following command.
+```bash
+cat /etc/upstream-release/lsb-release
+```
 
-A few other common libraries are tools also required such as openssl 1.1+, libcurl, curl, libusb, GMP, Python 3, and zlib.
+</details>
 
-**A Warning On Parallel Compilation Jobs (`-j` flag)**: When building C/C++ software often the build is performed in parallel via a command such as `make -j $(nproc)` which uses the number of CPU cores as the number of compilation jobs to perform simultaneously. However, be aware that some compilation units (.cpp files) in Leap are extremely complex and will consume nearly 4GB of memory to compile. You may need to reduce the level of parallelization depending on the amount of memory on your build host. e.g. instead of `make -j $(nproc)` run `make -j2`. Failures due to memory exhaustion will typically but not always manifest as compiler crashes.
+Once you have a `*.deb` file downloaded for your version of Ubuntu, you can install it as follows.
+```bash
+sudo apt-get update
+sudo apt-get install -y ~/Downloads/leap*.deb
+```
+Your download path may vary.
+- If you are in an Ubuntu docker container, omit `sudo` because you run as `root` by default.
 
-Generally we recommend performing what we refer to as a "pinned build" which ensures the compiler and boost version remain the same between builds of different Leap versions (Leap requires these versions remain the same otherwise its state needs to be repopulated from a portable snapshot).
+Finally, verify Leap was installed correctly.
+```bash
+nodeos --full-version
+```
+You should see a [semantic version](https://semver.org) string print out followed by a `git` commit hash with no errors. For example:
+```
+v3.1.2-0b64f879e3ebe2e4df09d2e62f1fc164cc1125d1
+```
 
-#### Building Pinned Build Binary Packages
-In the directory `<leap src>/scripts` you will find the two scripts `install_deps.sh` and `pinned_build.sh`. If you haven't installed build dependencies then run `install_deps.sh`. Then run `pinned_build.sh <dependencies directory> <leap build directory> <number of jobs>`.
+## Source Installation
```

Comment:
I agree, we should have a single source of truth. I don't like that there wouldn't be build instructions in the `/docs` folder, but I do think it needs to be in the `README.md`. That's where people are going to look. I wish there were a way to embed markdown documents in other markdown documents. We could link to the build instructions from the `README.md`, but I think it would be better for them to live in the `README.md`.

Reply:
Another option, to avoid wholesale duplication, is for the README to be more on the concise side -- a "quick start" -- and then linking to docs for more in depth details. (The README should probably not solely be installation and building instructions as it is today, but also at least links to more resources like the doc portal, discord, etc)

Reply:
Yes indeed, we could add those instructions back to `/docs` later once the "install from binaries" or "build from source" sections mature within the various README's (including `cdt`, `DUNE`, and `eos-system-contracts` at the ENF org). For now, we can keep those sections in the `/docs` but have small paragraphs that simply link to the corresponding sections in the README.

Reply:
> Another option, to avoid wholesale duplication, is for the README to be more on the concise side -- a "quick start" -- and then linking to docs for more in depth details.

I agree with this. Perhaps we just have binary install and pinned build script instructions here with a link to docs for unpinned builds.

> The README should probably not solely be installation and building instructions as it is today, but also at least links to more resources like the doc portal, discord, etc

Honestly, I found this weird when I started this ticket. Why is this just build and install instructions? I do think those belong in a `README.md` personally, but we only have one sentence explaining what this repo even is.

> Yes indeed, we could add those instructions back to /docs later once the "install from binaries" or "build from source" sections mature within the various README's (including cdt, DUNE, and eos-system-contracts at the ENF org). For now, we can keep those sections in the /docs but have small paragraphs that simply link to the corresponding sections in the README.

This seems to contradict itself. I like the first half, personally. I can copy/pasta the updated instructions into `/docs` and we'll have two copies, or delete the build doc altogether and we'll have one copy.

I am not opposed to moving the build instructions into the `/docs` folder. However, when this story started, there were _already_ two copies of the build instructions. It isn't clear which one the customer was using, but it is beginning to feel like unifying them is a bigger discussion with a lot of stakeholders that requires some strategy and should probably be a separate ticket from the one this pull request intends to address.

For example, the build instructions make sense in the document you linked. However, the stuff I wrote up about test suites really doesn't. How to run the tests, maybe, but descriptions of the test suites is a whole other thing that deserves a document in its own right. Test suites aren't about building. I also think some summary of the test suites (like the table) would belong in a succinct `README.md`.

I suggest we update the build instructions in both places for now, then open a new ticket to track structural changes to the documentation.

Reply:
Ah I meant keeping the section titles in the `/docs` (not the content) with small paragraphs linking to the corresponding README sections. I'd say don't worry about the build or install instructions in the `/docs` and just focus on updating the README. I also brought up those instructions in the `/docs` to make you aware that there's applicable content there that might need to be reconciled with the instructions you're writing/fixing in the README.

Reply:
> Ah I meant keeping the section titles in the /docs (not the content) with small paragraphs linking to the corresponding README sections.

Done.